end-to-end-test.sh

```diff
@@ -129,11 +144,20 @@ EOF
 
   if [ ${keep} -eq 0 ]
   then
-    kill -9 "$(cat ${tmpdir}/node_exporter.pid)"
+    kill "$(cat ${tmpdir}/node_exporter.pid)"
     # This silences the "Killed" message
     set +e
     wait "$(cat ${tmpdir}/node_exporter.pid)" > /dev/null 2>&1
+    rc=0
+    if [ ${socket} -ne 0 ]; then
+      ls -l "${unix_socket}" &> /dev/null
+      if [ $? -eq 0 ]; then
```

Comment:
This can be made a single line:

    if ls -l "${unix_socket}" &> /dev/null; then